curriculum/challenges/english/25-front-end-development/lab-color-picker/67bf4350777ac6ffdc027745.md

```diff
@@ -0,0 +1,320 @@
+const reactInstance = definitelyAnInput[key]
+assert.exists(reactInstance);
+assert.isFunction(reactInstance.onChange);
+```
+
+Your `#color-input` should have an initial value of `#ffffff`.
+
+```js
+const mostCertainlyAnInput = document.getElementById("color-input");
+assert.exists(mostCertainlyAnInput);
+assert.strictEqual(mostCertainlyAnInput.value, "#ffffff");
+```
+
+The initial value of your `#color-input` should be managed by your `useState` hook.
+
+```js
+const declaredState = code.match(/const\s*\[\s*(.*)\s*,.*\]\s*=\s*useState/);
+console.log(declaredState);
+const declaredStateVar = declaredState?.[1];
+console.log(declaredStateVar);
+const regex = new RegExp(`value\s*=\s*\{\s*${declaredStateVar}\s*}`);
+assert.match(code, regex);
+```
+
+When your `#color-input` value is changed, your `#color-picker-container` should change its background color to reflect that value.
+
+```js
+const inputForSure = document.getElementById("color-input");
+assert.exists(inputForSure);
+const nativeValueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;
+nativeValueSetter.call(inputForSure, "#000001");
+inputForSure.dispatchEvent(new Event("input", { bubbles: true}));
+const container = document.getElementById("color-picker-container");
+assert.exists(container);
+const style = window.getComputedStyle(container);
+assert.equal(style?.backgroundColor, "rgb(0, 0, 1)");
+```
+
+# --seed--
+
+## --seed-contents--
+
+```html
+<!DOCTYPE html>
+<html>
+
+<head>
+    <meta charset="UTF-8" />
+    <title>Color Picker</title>
+    <link rel="stylesheet" href="styles.css" />
+    <script src="https://unpkg.com/react@18/umd/react.development.js"></script>
+    <script src="https://unpkg.com/react-dom@18/umd/react-dom.development.js"></script>
+    <script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
+    <script
+      data-plugins="transform-modules-umd"
+      type="text/babel"
+      src="index.jsx"
+    ></script>
+</head>
+
+<body>
+    <div id="root"></div>
+    <script
+      data-plugins="transform-modules-umd"
+      type="text/babel"
+      data-presets="react"
+      data-type="module"
+    >
+      import { MyApp } from './index.jsx';
+      ReactDOM.createRoot(document.getElementById('root')).render(<MyApp />);
+    </script>
+</body>
+
+</html>
+```
+
+```css
+body,
+html {
+    margin: 0;
+    padding: 0;
+    height: 100%;
+    font-family: Arial, sans-serif;
+}
+
+#color-picker-container {
+    height: 100vh;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    flex-direction: column;
+    background-color: #ffffff;
+}
+
+#pick-color-btn {
+    padding: 15px 30px;
+    font-size: 18px;
+    color: white;
+    background-color: #14183c;
+    border: none;
+    border-radius: 5px;
+    cursor: pointer;
+    margin-top: 20px;
+}
+
+#pick-color-btn:hover {
+    background-color: #08131f;
+}
+
```

Comment:
None of these are mentioned in the user stories. So, including them does not help much.

```suggestion
```